src/org/jetbrains/plugins/scala/codeInsight/intention/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntention.scala

```diff
@@ -55,12 +62,100 @@ class ReplaceEqualsOrEqualityInMethodCallExprIntention extends PsiElementBaseInt
     val oper = methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.getText
 
     expr.append(methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].
-            qualifier.get.getText).append(".").append(replaceOper(oper)).append(methodCallExpr.args.getText)
+      qualifier.get.getText).append(".").append(replaceOper(oper)).append(methodCallExpr.args.getText)
 
     val newMethodCallExpr = ScalaPsiElementFactory.createExpressionFromText(expr.toString(), element.getManager)
 
     val size = newMethodCallExpr.asInstanceOf[ScMethodCall].getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.
+      getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+
+    inWriteAction {
+      methodCallExpr.replace(newMethodCallExpr)
+      editor.getCaretModel.moveToOffset(start + size)
+      PsiDocumentManager.getInstance(project).commitDocument(editor.getDocument)
+    }
+  }
+
+  def invoke_method_one(project: Project, editor: Editor, element: PsiElement) {
+    val methodCallExpr: ScMethodCall = PsiTreeUtil.getParentOfType(element, classOf[ScMethodCall], false)
+    if (methodCallExpr == null || !methodCallExpr.isValid) return
+
+    val start = methodCallExpr.getTextRange.getStartOffset
+
+    val replaceOper = Map("equals" -> "==", "==" -> "equals")
+
+    val args = new StringBuilder().append(methodCallExpr.args.getText)
+
+    val oper = methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.getText
+    val desiredOper: String = replaceOper(oper)
+
+    val expr = new StringBuilder()
+      .append(methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].qualifier.get.getText)
+
+    if (desiredOper == "==") {
+      expr.append(" ")
+      IntentionUtils.analyzeMethodCallArgs(methodCallExpr.args, args)
+      expr.append(desiredOper).append(" ").append(args)
+    } else {
+      expr.append(".")
+      expr.append(desiredOper).append(args)
+    }
+
+    print(expr.toString())
+    val newMethodCallExpr = ScalaPsiElementFactory.createExpressionFromText(expr.toString(), element.getManager)
+
+    val size = {
+      newMethodCallExpr match {
+        case scMethodCall: ScMethodCall =>
+          newMethodCallExpr.asInstanceOf[ScMethodCall].getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.
             getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+        case scInfixExpr: ScInfixExpr =>
+          newMethodCallExpr.asInstanceOf[ScInfixExpr].operation.nameId.getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+      }
+    }
+
+    inWriteAction {
+      methodCallExpr.replace(newMethodCallExpr)
+      methodCallExpr.replaceExpression(newMethodCallExpr, removeParenthesis = true)
```

Comment:
`replaceExpression` is more specialized. It works only for `ScExpression` and may remove/add enclosing parentheses. By the way, you can add a test for `x.equals(y).toString` example from the discussion of SCL-9579 where parentheses are needed.